lib/streamlit/elements/lib/mutable_status.py

```diff
@@ -0,0 +1,164 @@
+            parent._active_dg._cursor.delta_path if parent._active_dg._cursor else []
+        )
+
+        status_container = cast(
+            MutableStatus,
+            parent._block(block_proto=block_proto, dg_type=MutableStatus),
+        )
+
+        status_container._delta_path = delta_path
+        status_container._last_proto = block_proto
+        status_container._last_state = state
+        if expanded == "auto":
+            status_container._auto_collapse = True
+
+        # TODO: We cannot use this since it won't support the usage without context manager:
+        # with status_container:
+        #     try:
+        #         yield status_container
+        #     except Exception as ex:
+        #         status_container.update(state="error")
+        #         raise ex
+        return status_container
+
+    def __init__(
+        self,
+        root_container: int | None,
+        cursor: Cursor | None,
+        parent: DeltaGenerator | None,
+        block_type: str | None,
+    ):
+        super().__init__(root_container, cursor, parent, block_type)
+
+        # Initialized in `_create()`:
+        self._last_proto: BlockProto | None = None
+        self._last_state: States | None = None
+        self._delta_path: List[int] | None = None
+        self._auto_collapse: bool = False
+
+    def update(
+        self,
+        *,
+        label: str | None = None,
+        state: States | None = None,
+        expanded: bool | None = None,
+    ) -> None:
+        assert self._last_proto is not None, "Status not correctly initialized!"
+        assert self._delta_path is not None, "Status not correctly initialized!"
+
+        msg = ForwardMsg()
+        msg.metadata.delta_path[:] = self._delta_path
+        msg.delta.add_block.CopyFrom(self._last_proto)
+
+        if expanded is not None:
+            msg.delta.add_block.expandable.expanded = expanded
+
+        if label is not None:
+            msg.delta.add_block.expandable.label = label
+
+        if state is not None:
+            if state == "running":
+                msg.delta.add_block.expandable.icon = "spinner"
+            elif state == "complete":
+                msg.delta.add_block.expandable.icon = "check"
+            elif state == "error":
+                msg.delta.add_block.expandable.icon = "error"
+            self._last_state = state
+
+        self._last_proto = ForwardMsg().delta.add_block
+        self._last_proto.CopyFrom(msg.delta.add_block)
+
+        if expanded is None:
+            msg.delta.add_block.expandable.ClearField("expanded")
+
+        _enqueue_message(msg)
+
+    def __enter__(self) -> MutableStatus:  # type: ignore[override]
+        # This is a little dubious: we're returning a different type than
+        # our superclass' `__enter__` function. Maybe DeltaGenerator.__enter__
+        # should always return `self`?
+        super().__enter__()
+        if self._last_state != "running":
+            self.update(state="running")
```

Comment:
I noted in the spec but I think we should consider remove this, did you have something in mind for needing it?

I don't expect to see the spinner if I do something like this, but it blips up for a split second:

```python
with st.status("Doing some work", state="complete"):
    st.write("Did it!")
```

Reply:
I removed it 👍  I don't remember the reason why I added this 🙈 but I agree that we probably don't need this